src/context/WorkoutContext.tsx

```diff
@@ -0,0 +1,249 @@
+import { createContext, useContext, useEffect, useState } from 'react';
+
+// Timer configuration types
+export interface TimerConfig {
+    id: string;
+    type: 'stopwatch' | 'countdown' | 'tabata' | 'xy';
+    workTime: { minutes: number; seconds: number };
+    restTime?: { minutes: number; seconds: number }; // Optional, for Tabata and XY
+    totalRounds?: number; // Optional, for Tabata and XY
+    // currentRound?: number; // Optional, for Tabata and XY
+    timerMode?: 'work' | 'rest'; // Optional, for Tabata and XY
+    state: 'not running' | 'running' | 'paused' | 'completed';
+    skipped?: boolean; // Track if the timer was skipped
+}
+
+// Context State
+interface WorkoutContextState {
+    timers: TimerConfig[];
+    currentTimerIndex: number | null; // Index of the active timer
+    elapsedTime: number; // Elapsed time in milliseconds for active timer
```

Comment:
I'd call this `currentTimerElapsedtime` or something a little more informative, although the comment is a great aid here.